File: GithubUserJava/app/src/main/java/com/example/myapplication/factory/SettingViewModelFactory.java
package com.example.myapplication.factory;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import com.example.myapplication.data.local.datastore.SettingPreferences;
import com.example.myapplication.viewmodel.SettingViewModel;

public class SettingViewModelFactory implements ViewModelProvider.Factory {

    private final SettingPreferences pref;

    public SettingViewModelFactory(SettingPreferences pref) {
        this.pref = pref;
    }

    @SuppressWarnings("unchecked")
    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        if (modelClass.isAssignableFrom(SettingViewModel.class)) {
            return (T) new SettingViewModel(pref);
        }
        throw new IllegalArgumentException("Unknown ViewModel class: " + modelClass.getName());
    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/factory/SettingViewModelFactory.kt
package com.example.myapplication.factory

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.myapplication.data.local.datastore.SettingPreferences
import com.example.myapplication.viewmodel.SettingViewModel

class SettingViewModelFactory(private val pref: SettingPreferences) :
    ViewModelProvider.NewInstanceFactory() {
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(SettingViewModel::class.java)) {
            return SettingViewModel(pref) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class: " + modelClass.name)
    }
}